src/utils/vcs.rs

```diff
@@ -207,6 +207,13 @@ impl VcsUrl {
     }
 }
 
+fn extract_provider_name(host: &str) -> String {
```

Comment:
```suggestion
fn extract_provider_name(host: &str) -> &str {
```

[optional] I'd suggest returning a `&str` here, and performing the copy to `String` in the calling code if it is needed. There is no reason why this code needs to copy the extracted slice, even if now we need a copy, in the future we may want to use this function, where we do not need a copy.